pkg/util/log/log.go

```diff
@@ -288,13 +288,13 @@ func log(logLevel seelog.LogLevel, bufferFunc func(), scrubAndLogFunc func(strin
 
 	isInnerNil := l.inner == nil
 
-	if !isInnerNil && l.shouldLog(logLevel) {
+	if isInnerNil {
+		addLogToBuffer(bufferFunc)
+	} else if l.shouldLog(logLevel) {
 		s := BuildLogEntry(v...)
-
 		scrubAndLogFunc(s)
-	} else if isInnerNil {
-		addLogToBuffer(bufferFunc)
 	}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏In all those functions you don't need the `isInnerNil` variable anymore since it's only used once (and its definition is pretty trivial)

Reply:
Fixed